asus_stylus.py

```diff
@@ -23,82 +25,97 @@
-                stylus_detected = 1
+            if stylus_detection_status == 0 and "Stylus" in line:
+                stylus_detection_status = 1
                 log.debug('Detect stylus from %s', line.strip())
 
-            if stylus_detected == 1:
+            # Found stylus, now searching for ids
+            if stylus_detection_status == 1:
                 if "S: " in line:
                     # search device id
-                    device_id=re.sub(r".*i2c-(\d+)/.*$", r'\1', line).replace("\n", "")
-                    log.debug('Set stylus device id %s from %s', device_id, line.strip())
+                    stylus_id = re.sub(r".*i2c-(\d+)/.*$", r'\1', line).replace("\n", "")
+                    device_ids.append(stylus_id)
+                    log.debug('Set stylus device id %s from %s', stylus_id, line.strip())
 
                 if "H: " in line:
-                    stylus = line.split("event")[1]
-                    stylus = stylus.split(" ")[0]
-                    stylus_detected = 2
-                    log.debug('Set stylus id %s from %s', stylus, line.strip())
+                    stylus_id = line.split("event")[1].split(" ")[0]
+                    styluses.append(stylus_id)
+                    log.debug('Set stylus id %s from %s', stylus_id, line.strip())
+                    stylus_detection_status = 0
 
-            # Stop looking if stylus have been found
-            if stylus_detected == 2:
-                break
+    # Stylus was found on at least one device
+    if len(styluses) > 0:
+        stylus_detection_status = 2;
 
-    if stylus_detected != 2:
+    # Stylus was not detected?
+    if stylus_detection_status != 2:
         tries -= 1
         if tries == 0:
-            if stylus_detected != 2:
-                log.error("Can't find stylus (code: %s)", stylus_detected)
-            if stylus_detected == 2 and not device_id.isnumeric():
-                log.error("Can't find device id")
+            if stylus_detection_status != 2:
+                log.error("Can't find stylus (code: %s)", stylus_detection_status)
+            for device_id in device_ids:
+                if stylus_detection_status == 2 and not device_id.isnumeric():
+                    log.error("Can't find device id")
             sys.exit(1)
     else:
         break
 
 
 # Start monitoring the stylus
-fd_t = open('/dev/input/event' + str(stylus), 'rb')
-d_t = Device(fd_t)
+stylus_devices = []
+for stylus in styluses:
+    file_io = open('/dev/input/event' + str(stylus), 'rb')
+    stylus_devices.append(Device(file_io))
 
 
 # Create a new device
-dev = Device()
-dev.name = "Asus Stylus"
-for key_mapping in layout.keys:
-    dev.enable(key_mapping[2])
-dev.enable(EV_SYN.SYN_REPORT)
-dev.enable(EV_MSC.MSC_SCAN)
+class StylusInterface():
+    def __init__(self, stylus):
+        self.stylus = stylus
+        
+        self.device = Device()
+        self.device.name = "Asus Stylus"
```

Comment:
@Safenein Has to be used a different name for each created virtual device. Each creates a line for example in command:

````
$ libinput debug-events
...
 -event11 DEVICE_ADDED **Asus Stylus** seat0 default group11 cap:p left scroll-nat
...
```

Reply:
@Safenein The added name you pushed is just a copy of an already existing virtual device name from the origin driver. Has to be added something like I above proposed Buttons or maybe better for identification something like **github.com/asus-linux-drivers** so together it will look like:

```
ELAN9009:00 04F3:2E36 Stylus (github.com/asus-linux-drivers)
ELAN9009:00 04F3:2E36 Stylus
```

I am open to ideas, I do not insist on this.

Reply:
It should be good with 2c9d253 (untested).

Reply:
@Safenein Works. Marked as resolved.